webrender/src/prim_store.rs

```diff
@@ -58,15 +58,38 @@ pub struct PrimitiveCacheInfo {
     pub key: PrimitiveCacheKey,
 }
 
+#[derive(Debug)]
+pub enum PrimitiveClipInfo {
+    NoClip,
+    Complex(Rect<f32>, f32),
+    Region(ClipRegion),
+    GpuAddress(GpuStoreAddress),
+}
+
+impl PrimitiveClipInfo {
+    pub fn is_clipped(&self) -> bool {
```

Comment:
Will this return true even for a rectangular clip only? (If so, that will prevent occlusion culling where it is being used).